src/media/js/apps.js

```diff
@@ -113,10 +113,59 @@ define('apps',
         if (product.payment_required && !product.price) {
             reasons.push(gettext('not available for your region'));
         }
-        if (!product.isWebsite && !capabilities.webApps ||
-            (!capabilities.packagedWebApps && product.is_packaged) ||
-            !_.contains(product.device_types, device)) {
-            reasons.push(gettext('not available for your platform'));
+        if (!capabilities.webApps &&
+            !_.contains(product.device_types, 'firefoxos') &&
+            !_.contains(product.device_types, 'android-mobile') &&
+            !_.contains(product.device_types, 'android-tablet') &&
+            _.contains(product.device_types, 'desktop')) {
+            reasons.push(gettext('try this app on Firefox Desktop'));
+        }
+        if (!capabilities.firefoxAndroid &&
+            !_.contains(product.device_types, 'desktop') &&
+            !_.contains(product.device_types, 'firefoxos') &&
+            (_.contains(product.device_types, 'android-mobile') ||
+            _.contains(product.device_types, 'android-tablet'))) {
+            reasons.push(gettext('try this app on Android'));
+        }
+        if (!capabilities.firefoxOS &&
+            !_.contains(product.device_types, 'desktop') &&
+            !_.contains(product.device_types, 'android-mobile') &&
+            !_.contains(product.device_types, 'android-tablet') &&
+            _.contains(product.device_types, 'firefoxos')) {
+            reasons.push(gettext('try this app on Firefox OS'));
```

Comment:
hey @ddurst & @diox
@diox thanks for simplifying it :) 
I tried to simplify it as much I can, but can't ignore these "desktop/android", "android/firefox os", "desktop/firefox os". But this has been simplified than in the commit in this pr. So, here, is what I want to write- 
```
/* 
 * Return message to show the client when a product is not compatible, trying
 * to find a compatible device to give the user an actionable suggestion.
 */
function getIncompatibilityMessage(product) {
    var message;

    if (_.contains(product.device_types, 'firefoxos') && 
       (_.contains(product.device_types, 'android-tablet') || 
        _.contains(product.device_types, 'android-mobile')) && 
        _.contains(product.device_types, 'desktop')) {
        message = gettext('try this app on Firefox');
    } else if (_.contains(product.device_types, 'firefoxos') && 
                 (_.contains(product.device_types, 'android-tablet') || 
                  _.contains(product.device_types, 'android-mobile'))) {
        message = gettext('available on Android & Firefox OS');
    } else if ((_.contains(product.device_types, 'android-tablet') || 
                  _.contains(product.device_types, 'android-mobile')) && 
                 _.contains(product.device_types, 'desktop')) {
        message = gettext('available on Desktop & Andorid');
    } else if (_.contains(product.device_types, 'firefoxos') && 
                 _.contains(product.device_types, 'desktop')) {
        message = gettext('available on Firefox Desktop & Firefox OS');
    } else if (_.contains(product.device_types, 'firefoxos')) {
        message = gettext('try this app on Firefox OS');
    } else if (_.contains(product.device_types, 'android-mobile')) {
        message = gettext('try this app on Firefox for Android Mobile');
    } else if (_.contains(product.device_types, 'android-tablet')) {
        message = gettext('try this app on Firefox for Android Tablet');
    } else if (_.contains(product.device_types, 'desktop')) {
        message = gettext('try this app on Firefox Desktop');
    }
    return message;
}

if (!product.isWebsite) {
    // The client is missing some capabilities.
    if (!capabilities.webApps || 
        (!capabilities.packagedWebApps && product.is_packaged) || 
         !_.contains(product.device_types, device)) {
        reasons.push(getIncompatibilityMessage(product));
    };
}
```
Please check if it is enough simplified :)